### PR TITLE
Implement multiple response content types

### PR DESCRIPTION
**Pull request by bterlson**
I was hoping to implement multiple REQUEST bodies to unblock Will but that side is much harder and seems wrapped up in the metadata proposal. Multiple response bodies though was fairly easy.

### CODE REVIEW

File: packages/openapi3/src/lib.ts [hunk 1]
```diff
@@ -30,7 +30,13 @@ export const libDef = {
     "duplicate-response": {
       severity: "error",
       messages: {
-        default: paramMessage`Multiple return types for status code ${"statusCode"}`,
+        default: paramMessage`Multiple return types for content type ${"contentType"} and status code ${"statusCode"}`,
```
Comment (timotheeguerin) [inline packages/openapi3/src/lib.ts:33]: Not for this PR but what we were talking about: Should that validation technically be moved to the rest layer?

File: packages/openapi3/src/openapi.ts [hunk 2]
```diff
@@ -424,20 +424,46 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
...
+      }
+
       if (Object.keys(headers).length > 0) {
-        response.headers = headers;
+        response.headers ??= {};
+
+        // OpenAPI can't represent different headers per content type.
+        // So we merge headers here, and report any duplicates.
+        // It may be possible in principle to not error for identically declared
+        // yeaders.
```
Comment (timotheeguerin) [inline packages/openapi3/src/openapi.ts:452]: typo
```suggestion
        // headers.
```

File: packages/openapi3/src/openapi.ts [hunk 3]
```diff
@@ -424,20 +424,46 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
...
+        for (const [key, value] of Object.entries(headers)) {
+          if (response.headers[key]) {
```
Comment (timotheeguerin) [inline packages/openapi3/src/openapi.ts:454]: Without ignoring headers that are the same how could one represent a common header or would it just not be possible yet.

**Review by timotheeguerin — APPROVED** [review]

File: packages/openapi3/src/openapi.ts (hunk 3 above)
Comment (bterlson) [reply packages/openapi3/src/openapi.ts:454]: I don't think you could. But I didn't want to do much here in advance of the metadata proposal.

Comment (bterlson) [reply packages/openapi3/src/openapi.ts:454]: (but I also felt that the story depending on declaring identical headers across all requests felt not so good, e.g. users might be surprised that headers from one content type affect headers for another content type).

Comment (timotheeguerin) [reply packages/openapi3/src/openapi.ts:454]: yeah that's true

**Pull request merged**